crates/vim/src/normal/delete.rs

```diff
@@ -33,12 +34,20 @@ impl Vim {
                     s.move_with(|map, selection| {
                         let original_head = selection.head();
                         original_columns.insert(selection.id, original_head.column());
-                        let kind =
+                        let mut kind =
                             motion.expand_selection(map, selection, times, &text_layout_details);
 
                         ranges_to_copy
                             .push(selection.start.to_point(map)..selection.end.to_point(map));
 
+                        if inclusive {
```

Comment:
any reason this needs to go below `ranges_to_copy`? If not, I think you could avoid `mut` with:

```
let kind = match motion.expand_selection(...) {
  Some(x) => Some(x)
  None => {
     if (inclusive) {
       {selection mutating code}
       Some(MotionKind::Inclusive)
     } else {
        None
     }
  }
}
```

But no strong opinion on that, depending on if you think that reads worse.

Reply:
To do it that way the code would look closer to 
```
let kind = match motion.expand_selection(...) {
  Some(x) if inclusive => {
    {selection mutation code}
    Some(x)
  },
  None if inclusive => {
       {selection mutating code}
       Some(MotionKind::Inclusive)
  },
  Some(x) => Some(x),
  None => None
}
```
I split the difference and was able to keep the kind from being a `mut` but the mutation code happens after the fact.